src/main/webapp/app/entities/quiz-exercise/quiz-exercise-detail.component.ts

```diff
@@ -115,25 +133,25 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, OnDestroy
         const today = moment();
         this.minDate = { year: today.year(), month: today.month() + 1, day: today.date() };
 
-        this.paramSub = this.route.params.subscribe(params => {
-            /** Query the courseService for the participationId given by the params */
-            if (params['courseId']) {
-                this.courseService.find(params['courseId']).subscribe((response: HttpResponse<Course>) => {
-                    this.course = response.body;
-                    // Make sure to call init if we didn't receive an id => new quiz-exercise
-                    if (!params['id']) {
-                        this.init();
-                    }
-                });
-            }
-            if (params['id']) {
-                this.quizExerciseService.find(params['id']).subscribe((response: HttpResponse<QuizExercise>) => {
-                    this.quizExercise = response.body;
-                    this.courseRepository = this.courseService;
+        const courseId = +this.route.snapshot.paramMap.get('courseId');
+        const quizId = +this.route.snapshot.paramMap.get('id');
```

Comment:
- if courseId is not null and quizId is not null, the courseService is triggered without any effect. Therefore the first condition should be courseId && !quizId
-  also both conditions seem to be exclusive to me: if (courseId && !quizId)... else if(quizId)

Reply:
I made a commit - this change was based on a false assumption.
I noticed this when testing, the first call is also used to assign the course to the component. Therefore closed.